MaxText/tests/inference/kvcache_test.py

```diff
@@ -107,25 +105,22 @@ def test_update_kv_cache(self):
     model_mode = MODEL_MODE_AUTOREGRESSIVE
     low_rank_main_1 = jnp.ones((self.batchsize, 1, self.kv_lora_rank), dtype=self.dtype) * 0.04
     key_rope_1 = jnp.ones((self.batchsize, 1, 1, self.k_rope_head_dim), dtype=self.dtype) * 0.05
-    _, new_vars = test_module.apply(
-        new_vars,
+    _ = test_module(
         low_rank_main_1,
         key_rope_1,
         decoder_segment_ids,
         model_mode,
-        rngs={"params": random.PRNGKey(0)},
-        mutable=True,
```

Comment:
@cgarciae looks like we need to continue passing `model_mode` into the __call__ functions based on this unit test. The `MlaKVCache` is initialized once but called with different `model_mode` values

@mitalisi @patemotter @jrplatin let us know if there is a way to set `model_mode` as a self variable in __init__. It would help when we remove the `KVCache` Linen bridges